Gemfile.lock

```diff
@@ -8,84 +21,90 @@ PATH
       jquery-rails (~> 4.2)
       kaminari (~> 1.0, >= 1.2.1)
-      railties (>= 6.0, < 6.2)
+      railties (>= 6.0, < 7.1)
       ransack (~> 2.1, >= 2.1.1)
 
 GEM
   remote: https://rubygems.org/
   specs:
-    actioncable (6.1.4.4)
-      actionpack (= 6.1.4.4)
-      activesupport (= 6.1.4.4)
+    actioncable (7.0.1)
+      actionpack (= 7.0.1)
+      activesupport (= 7.0.1)
       nio4r (~> 2.0)
       websocket-driver (>= 0.6.1)
-    actionmailbox (6.1.4.4)
-      actionpack (= 6.1.4.4)
-      activejob (= 6.1.4.4)
-      activerecord (= 6.1.4.4)
-      activestorage (= 6.1.4.4)
-      activesupport (= 6.1.4.4)
+    actionmailbox (7.0.1)
+      actionpack (= 7.0.1)
+      activejob (= 7.0.1)
+      activerecord (= 7.0.1)
+      activestorage (= 7.0.1)
+      activesupport (= 7.0.1)
       mail (>= 2.7.1)
-    actionmailer (6.1.4.4)
-      actionpack (= 6.1.4.4)
-      actionview (= 6.1.4.4)
-      activejob (= 6.1.4.4)
-      activesupport (= 6.1.4.4)
+      net-imap
+      net-pop
+      net-smtp
+    actionmailer (7.0.1)
+      actionpack (= 7.0.1)
+      actionview (= 7.0.1)
+      activejob (= 7.0.1)
+      activesupport (= 7.0.1)
       mail (~> 2.5, >= 2.5.4)
+      net-imap
+      net-pop
+      net-smtp
       rails-dom-testing (~> 2.0)
-    actionpack (6.1.4.4)
-      actionview (= 6.1.4.4)
-      activesupport (= 6.1.4.4)
-      rack (~> 2.0, >= 2.0.9)
+    actionpack (7.0.1)
+      actionview (= 7.0.1)
+      activesupport (= 7.0.1)
+      rack (~> 2.0, >= 2.2.0)
       rack-test (>= 0.6.3)
       rails-dom-testing (~> 2.0)
       rails-html-sanitizer (~> 1.0, >= 1.2.0)
-    actiontext (6.1.4.4)
-      actionpack (= 6.1.4.4)
-      activerecord (= 6.1.4.4)
-      activestorage (= 6.1.4.4)
-      activesupport (= 6.1.4.4)
+    actiontext (7.0.1)
+      actionpack (= 7.0.1)
+      activerecord (= 7.0.1)
+      activestorage (= 7.0.1)
+      activesupport (= 7.0.1)
+      globalid (>= 0.6.0)
       nokogiri (>= 1.8.5)
-    actionview (6.1.4.4)
-      activesupport (= 6.1.4.4)
+    actionview (7.0.1)
+      activesupport (= 7.0.1)
       builder (~> 3.1)
       erubi (~> 1.4)
       rails-dom-testing (~> 2.0)
       rails-html-sanitizer (~> 1.1, >= 1.2.0)
-    activejob (6.1.4.4)
-      activesupport (= 6.1.4.4)
+    activejob (7.0.1)
+      activesupport (= 7.0.1)
       globalid (>= 0.3.6)
-    activemodel (6.1.4.4)
-      activesupport (= 6.1.4.4)
+    activemodel (7.0.1)
+      activesupport (= 7.0.1)
     activemodel-serializers-xml (1.0.2)
       activemodel (> 5.x)
       activesupport (> 5.x)
       builder (~> 3.1)
-    activerecord (6.1.4.4)
-      activemodel (= 6.1.4.4)
-      activesupport (= 6.1.4.4)
-    activerecord-jdbc-adapter (61.0-java)
-      activerecord (~> 6.1.0)
-    activerecord-jdbcsqlite3-adapter (61.0-java)
-      activerecord-jdbc-adapter (= 61.0)
+    activerecord (7.0.1)
+      activemodel (= 7.0.1)
+      activesupport (= 7.0.1)
+    activerecord-jdbc-adapter (50.0)
```

Comment:
Downgrade not expected. There are no Rails 7.0 compatible releases

Reply:
Indeed! I guess this is because this old version didn't specify a requirement on Rails, so bundler "correctly" resolves to that.

I guess the best thing we can do is fork the repository, relax the Rails requirement to allow Rails 7, point to that fork for now and let it fail in whatever way it fails. Ideally it doesn't fail at all and we can create a PR upstream to relax the requirement. Actually that PR can be probably created right away, so that people can start messing with their master branch and trying to add support.

Reply:
I've noticed that `activerecord-jdbc-adapter`'s approach would be moving the current `master` branch to `61-stable` and start committing Rails 7 related PRs to `master`. I'm not confident in testing against JRuby or changing the default CI build to "7-0-stable", should I open an issue instead?

Reply:
Last time I went ahead and created a PR directly: https://github.com/jruby/activerecord-jdbc-adapter/pull/1076, I think doing that should be fine, but if you want to gather some feedback from the repository maintainers first, feel free to open an issue 👍.

Reply:
Maybe this is locked upstream because of jruby/jruby#6464. As far as I can tell, there are no JRuby stable versions compatible with Rails 7

Reply:
Indeed! I think it still makes sense to propose the update to `activerecord-jdbc-adapter`. It allows us to at least have a proper "Rails 7 compatible" resolution on JRuby, and it allows testing Rails 7 support on jruby-head.